pkg/virt-handler/vm.go

```diff
@@ -1907,9 +1907,7 @@ func (c *VirtualMachineController) defaultExecute(key string,
 			log.Log.Object(vmi).V(3).Info("Deleting domain for VirtualMachineInstance with deletion timestamp.")
 			shouldDelete = true
 		default:
-			if vmi.IsFinal() {
-				shouldCleanUp = true
-			}
```

Comment:
I think it makes sense to always cleanup if the vmi is deleted, however curious regarding to how this could happen. Why would a deleted VMI not be final?

Reply:
A vmi can be in scheduling/scheduled as well.